src/V3Gate.cpp

```diff
@@ -121,6 +121,11 @@ class GateEitherVertex VL_NOT_FINAL : public V3GraphVertex {
     }
 };
 
+template <>
+bool V3GraphVertex::privateTypeTest<GateEitherVertex>(const V3GraphVertex* vtxp) {
+    return dynamic_cast<const GateEitherVertex*>(vtxp);
+}
+
```

Comment:
I removed this code just to see what compiler error we would get and everything passed???  Similar on some others.

Reply:
This exact specialization is not called anywhere, i.e. there's no `is<GateEitherVertex>()`. If you actually call it the linker fails.

This works like this because declarations of symbols from other code units (and libs), and even their use, are totally normal for a compiler. A linker is responsible for reporting missing definitions.
However, a compiler doesn't just put references to unused declarations in an output object file, so a linker is not aware such declarations exist. As a result there are no warnings.

Anyway, your comment made me realize that my last commit introduced an error - the code in astgen doesn't detect namespaces, and I've put a class in a namespace...

Reply:
Perhaps have astgen emit C++ with a 'is' test for every vertex type to find these?